### PR TITLE
Added documentation for running discosrv behind nginx reverse proxy

### PR DESCRIPTION
**Pull request by firecat53**
Added documentation for running the discovery server behind nginx reverse proxy as requested by @AudriusButkevicius. Hopefully this is what you were looking for. Please let me know if I can change anything (or made any dumb mistakes).

Scott

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
So I think this should be written to assume that -http is there (potentially updating the docs as it seems to be missing), and not mention all the commits and versions as it's just confusing as it doesn't say which commit my version is from etc.

I think there should be a `Reverse Proxy Setup` section explaining that you need to run with -http option, and pass X-
SSL-Cert header containing PEM encoded client certificate and X-Forwarded-For header containing clients real IP.

Then in the nginx subsection you could talk how to provide the 2 headers and force client to provide certificates and giving the example config of yours and pointing to the forum link.

Something simillar to https://docs.syncthing.net/users/reverseproxy.html, where it explains what it is a bit, and what it does and why it's useful/better.

Also, not sure why you think the UDP port forwarding is still required?

File: users/discosrv.rst [hunk 1]
```diff
@@ -100,3 +100,67 @@ IP address, you will need a dynamic DNS service.
...
+use https://discovery.mydomain.com as the discovery server URL in the Syncthing
+settings::
+
+    upstream discovery.mydomain.com {
+        # Local IP address:port for discovery server
+        server 172.17.0.6:8443;
+    }
+    server {
+            server_name discovery.mydomain.com;
+            listen 80 ;
```
Comment (AudriusButkevicius) [inline users/discosrv.rst]: the extra space before ; on both listen lines is not needed.

File: users/discosrv.rst [hunk 2]
```diff
@@ -100,3 +100,67 @@ IP address, you will need a dynamic DNS service.
...
+            ssl_session_cache shared:SSL:50m;
+            ssl_certificate /etc/nginx/certs/discovery.mydomain.com.crt;
+            ssl_certificate_key /etc/nginx/certs/discovery.mydomain.com.key;
+            ssl_dhparam /etc/nginx/certs/discovery.mydomain.com.dhparam.pem;
+            add_header Strict-Transport-Security "max-age=31536000";
+            ssl_verify_client optional_no_ca;
+            location / {
+                    proxy_pass http://discovery.mydomain.com;
+            }
+    }
```
Comment (AudriusButkevicius) [inline users/discosrv.rst]: This is actually missing both X-Forwarded-For and X-SSL-Cert, so I suspect this doesn't work.

**Comment by firecat53** [comment]
Yipes...totally forgot to paste the proxy parameters...was in a hurry last night :(  I'll fix that up and work on the other suggestions. Thanks!

Scott

**Comment by calmh** [comment]
@st-review merge please

users/discosrv: Document running discosrv behind nginx

**Comment by st-review** [comment]
:ok_hand: Merged as 5ebce765249898169c66a226a686dc2b021400ad. Thanks, @firecat53!

**Pull request closed without merging**